schema/cassandra/cadence/schema.cql

```diff
@@ -260,13 +260,21 @@ CREATE TYPE task (
   partition_config map<text, text>
 );
 
+CREATE TYPE task_list_partition_config (
+  version              bigint,
+  num_read_partitions  int,
+  num_write_partitions int
+);
+
+
 CREATE TYPE task_list (
   domain_id        uuid,
   name             text,
   type             int, -- enum TaskRowType {ActivityTask, DecisionTask}
   ack_level        bigint, -- task_id of the last acknowledged message
   kind             int, -- enum TaskListKind {Normal, Sticky}
-  last_updated     timestamp
+  last_updated     timestamp,
+  adaptive_partition_config frozen<task_list_partition_config>
```

Comment:
I think we should make this a byte array representing a thrift serialized blob. This way further iterations on this adaptive_partition_config (e.g. adding zone info or other metadata) will not require DB schema change

Reply:
My argument is if we want to use serialized blob, we should do it for all.

Reply:
I think any future structs like this that we introduce going forward should be modeled as binary blob for flexibility purposes, therefore no more new DB types. Migrating away from existing DB types to blob approach is not P0 in my mind. We can consider a major schema at some point and that would the good time to get model everything as blob.